src/comm/TcpAcceptor.cc

```diff
@@ -318,6 +315,7 @@ Comm::TcpAcceptor::acceptNext()
     Must(IsConnOpen(conn));
     debugs(5, 2, HERE << "connection on " << conn);
     acceptOne();
+    SetSelect(conn->fd, COMM_SELECT_READ, doAccept, this, 0);
```

Comment:
Please move this SetSelect() call to the end of the acceptOne() method to

* avoid calling SetSelect() twice on Comm::NOMESSAGE and
* avoid calling SetSelect() at all after fatal accepting failures.